lib/node_modules/@stdlib/lapack/base/dlauu2/lib/base.js

```diff
@@ -0,0 +1,114 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isRowMajor = require( '@stdlib/ndarray/base/assert/is-row-major' );
+var dgemv = require( '@stdlib/blas/base/dgemv' ).ndarray;
+var dscal = require( '@stdlib/blas/base/dscal' ).ndarray;
+var ddot = require( '@stdlib/blas/base/ddot' ).ndarray;
+
+
+// MAIN //
+
+/**
+* Computes the product `U*U^T` or `L^T*L`, where the triangular factor `U` or `L` is stored in the upper or lower triangular part of array `A`.
+*
+* @private
+* @param {string} uplo - specifies whether to copy the upper or lower triangular/trapezoidal part of matrix `A`
+* @param {NonNegativeInteger} N - order of triangular factor `U` or `L`.
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index for `A`
+* @returns {integer} status code
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var out = dlauu2( 'upper', 2, A, 2, 1, 0 );
+* // returns 0
+*/
+function dlauu2( uplo, N, A, strideA1, strideA2, offsetA ) {
+	var aii;
+	var i;
+
+	// Quick return if possible
+	if ( N === 0 ) {
+		return 0;
+	}
+	if ( uplo === 'upper' ) {
+		// Compute the product U*U^T.
+		if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+			for ( i = 0; i < N; i++ ) {
+				aii = A[ offsetA + ( i * strideA1 ) + ( i * strideA2 ) ];
+				if ( i < N - 1 ) {
+					A[ offsetA + ( i * strideA1 ) + ( i * strideA2 ) ] = ddot( N - i, A, strideA2, offsetA + ( i * strideA1 ) + ( i * strideA2 ), A, strideA2, offsetA + ( i * strideA1 ) + ( i * strideA2 ) );
+					dgemv( 'no-transpose', i, N - i - 1, 1.0, A, strideA1, strideA2, offsetA + ( ( i + 1 ) * strideA2 ), A, strideA2, offsetA + ( ( i + 1 ) * strideA2 ) + ( i * strideA1 ), aii, A, strideA1, offsetA + ( i * strideA2 ) );
+				} else {
+					dscal( i + 1, aii, A, strideA1, offsetA + ( i * strideA2 ) );
+				}
+			}
+			return 0;
+		}
+		// column-major
+		for ( i = 0; i < N; i++ ) {
+			aii = A[ offsetA + ( i * strideA1 ) + ( i * strideA2 ) ];
+			if ( i < N - 1 ) {
+				A[ offsetA + ( i * strideA1 ) + ( i * strideA2 ) ] = ddot( N - i, A, strideA2, offsetA + ( i * strideA1 ) + ( i * strideA2 ), A, strideA2, offsetA + ( i * strideA1 ) + ( i * strideA2 ) );
+				dgemv( 'no-transpose', i, N - i - 1, 1.0, A, strideA1, strideA2, offsetA + ( ( i + 1 ) * strideA1 ), A, strideA1, offsetA + ( ( i + 1 ) * strideA1 ) + ( i * strideA2 ), aii, A, strideA2, offsetA + ( i * strideA1 ) );
+			} else {
+				dscal( i + 1, aii, A, strideA1, offsetA + ( i * strideA2 ) );
+			}
+		}
+		return 0;
+	}
+	// Lower
+	// Compute the product L^T*L.
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		for (i = 0; i < N; i++ ) {
+			aii = A[ offsetA + ( i * strideA1 ) + ( i * strideA2 ) ];
```

Comment:
@Pranavchiku You are repeatedly computing the same values over and over. You'll clarify and simplify the implementation if you use temporary variables. This applies here and to other PRs.